File: client/src/app/Models/WorkOrder.ts
import { Asset } from './Asset';
import { TaskStatus } from './TaskStatus';

export interface WorkOrder {
  id: number;
  companyId: number;
  locationId: number;
  internalNumber: number;
  taskName: string;
  taskDescription: string;
  startDate: string;
  dueDate: string;
  taskAssignmentId: number;
  estimatedTime: number;
  taskTypeId: number;
  assetId: number;
  completionDate: string;
  completionNote: string;
  priorityId: number;
  completionRatio: number;
  assetDownTime: number;
  taskStatus?: TaskStatus | null;
  asset?: Asset | null;
}

File: CAFM.Domain/Entities/WorkOrder.cs
namespace CAFM.Domain.Entities;

public class WorkOrder
{
    public long Id { get; set; }

    public int CompanyId { get; set; }

    public long LocationId { get; set; }

    public long InternalNumber { get; set; }

    public string TaskName { get; set; }

    public string TaskDescription { get; set; }

    public DateTime CreatedDate { get; set; }

    public DateTime? ModifiedDate { get; set; }

    public DateTime? StartDate { get; set; }

    public DateTime DueDate { get; set; }

    public long? TaskAssignmentId { get; set; }

    public int EstimatedTime { get; set; }

    public short TaskTypeId { get; set; }

    public long? AssetId { get; set; }

    public DateTime? CompletionDate { get; set; }

    public string CompletionNote { get; set; }

    public int PriorityId { get; set; }

    public int CompletionRatio { get; set; }

    public int TaskStatusId { get; set; }

    public int AssetDownTime { get; set; }

    public bool IsDeleted { get; set; }

    public long CreatedBy { get; set; }

    public virtual Asset Asset { get; set; }

    public virtual TaskPriority Priority { get; set; }

    public virtual TaskStatue TaskStatus { get; set; }

    public virtual ICollection<WorkOrderDetail> WorkOrderDetails { get; set; } = new List<WorkOrderDetail>();
}
